libff/algebra/field_utils/bigint.tcc

```diff
@@ -60,6 +62,42 @@ bigint<n>::bigint(const mpz_t r) /// Initialize from MPZ element
     mpz_clear(k);
 }
 
+template<mp_size_t n>
+bigint<n> bigint<n>::operator-(const bigint<n> &other) const
```

Comment:
I'm pretty confused about why this changed. Did most of the functions become non-mutative?

Reply:
These functions are needed to implement Tonelli-Shanks parameter creation, but if those parameters are hard-coded they shouldn't be needed.

Reply:
Does Tonelli-Shanks need big integer arithmetic (as opposed to field arithmetic)?

Reply:
Tonelli-Shanks itself does not, but to calculate the parameters you need for it you do. We'll usually just hard-code those parameters though, so these functions won't really be needed.

Reply:
If so, lets add some comments to indicate that these aren't used in the high performance field arithmetic code. (Still feels a bit weird to have these)

Reply:
race condition in comments!

Ok that makes sense, lets add some comments for that. (That way people reviewing field arithmetic know that this isn't done in runtime)

Reply:
Per slack discussion, we are removing this code, as it should be generated from Sage. 

Feel free to keep it around in a separate branch for potentially one day defining fields in run time.